test/000_fullchain.test.ts

```diff
@@ -297,10 +298,125 @@ describe('Integration tests', function () {
+                const winningWorkers = workersAvailable.slice(1, workerNumber);
+                let workers = [{ signer: worker1, resultDigest: badResultDigest }];
+                for (const worker of winningWorkers) {
+                    workers.push({ signer: worker, resultDigest: resultDigest });
+                }
+                const acounts = [
+                    requester,
+                    scheduler,
+                    appProvider,
+                    datasetProvider,
+                    losingWorker,
+                    ...winningWorkers,
+                ];
+                // Create deal.
+                const orders = buildOrders({
+                    assets: ordersAssets,
+                    prices: ordersPrices,
+                    requester: requester.address,
+                    tag: standardDealTag,
+                    beneficiary: beneficiary.address,
+                    volume,
+                    trust: winningWorkers.length,
+                });
+
+                const { dealId, dealPrice, schedulerStakePerDeal } =
+                    await iexecWrapper.signAndMatchOrders(...orders.toArray());
+                const taskPrice = appPrice + datasetPrice + workerpoolPrice;
+                const schedulerStakePerTask = schedulerStakePerDeal / volume;
+                const workerRewardPerTask = await iexecWrapper.computeWorkerRewardPerTask(
+                    dealId,
+                    PocoMode.CLASSIC,
+                );
+                const schedulerRewardPerTask = workerpoolPrice - workerRewardPerTask;
+                // Check initial balances.
+                let accountsInitBalances = [
+                    {
+                        address: proxyAddress,
+                        frozen: (await iexecPoco.frozenOf(proxyAddress)).toNumber(),
+                    },
+                ];
+                for (const account of acounts) {
+                    let address = account.address;
+                    let frozen = (await iexecPoco.frozenOf(account.address)).toNumber();
+                    accountsInitBalances.push({ address, frozen });
+                }
+                for (const worker of workers) {
+                    expect(await iexecPoco.viewScore(worker.signer.address)).to.be.equal(0);
                 }
+                const taskId = await iexecWrapper.initializeTask(dealId, 0);
+                // Finalize each task and check balance changes.
+                const workerStake = await iexecPoco
+                    .viewDeal(dealId)
+                    .then((deal) => deal.workerStake.toNumber());
+                for (const worker of workers) {
+                    await iexecWrapper.contributeToTask(
+                        dealId,
+                        0,
+                        worker.resultDigest,
+                        worker.signer,
+                    );
+                }
+                for (const winningWorker of winningWorkers) {
```

Comment:
You can also reveal for the losing worker and check that it fails.

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/160/commits/0764ba1e90e606906a52a302d8cf30cf1aba6072)